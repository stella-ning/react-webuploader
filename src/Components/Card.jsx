import React from 'react';
class Card extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
        let {title, extra} = this.props;
        const header = title
            ? <div className="wu-card-header">{title}</div>
            : null;
        return (
            <div className="wu-card-box">
                {header}
                <div className="wu-card-content">
                    {this.props.children}
                </div>
            </div>
        );
    }
}
export default Card;
