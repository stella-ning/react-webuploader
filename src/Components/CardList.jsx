import React from 'react';
import Card from './Card.jsx';
class CardList extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
        let {cardList, ...others} = this.props;
        return (
            <div className="wu-card-list">
                {cardList.map((element, index)=>(<Card {...element} key={index}></Card>))}
                {this.props.children}
            </div>
        );
    }
}
export default CardList;
