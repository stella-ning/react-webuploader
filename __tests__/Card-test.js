jest.unmock('../src/Components/Card.jsx');
jest.unmock('../src/Components/CardList.jsx');
jest.mock('react/lib/ReactDefaultInjection');

import React from 'react';
import ReactDOM from 'react-dom';
// import TestUtils from 'react-addons-test-utils';
import renderer from 'react-test-renderer';
import Card from '../src/Components/Card.jsx';
import CardList from '../src/Components/CardList.jsx';

describe('Card', () => {
    it('List', () => {
        const list = [{status: 'wu-upload-success', percent: null},{status: 'wu-upload-success', percent: null}];
        const component = renderer.create(
            <CardList width="" cardList={list}>
                <Card status={""} percent={null}>
                    <span onClick={()=>{}}>Add new</span>
                </Card>
            </CardList>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
