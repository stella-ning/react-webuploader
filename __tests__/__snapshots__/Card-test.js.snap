exports[`Card List 1`] = `
<div
  className="wu-card-list">
  <div
    className="wu-card-box wu-upload-success">
    <svg
      className="wu-card-bg"
      height={320}
      width={200}
      xmlns="http://www.w3.org/2000/svg">
      <path
        className="wu-card-progressbg"
        d="M 6 2 H 194 Q 198 2,198 6 V 314 Q 198 318,194 318 H 6 Q 2 318,2 314 V 6 Q 2 2,6 2 z" />
      <path
        className="wu-card-border"
        d="M 6 2 H 194 Q 198 2,198 6 V 314 Q 198 318,194 318 H 6 Q 2 318,2 314 V 6 Q 2 2,6 2 z" />
    </svg>
    <div
      className="wu-card-content" />
  </div>
  <div
    className="wu-card-box wu-upload-success">
    <svg
      className="wu-card-bg"
      height={320}
      width={200}
      xmlns="http://www.w3.org/2000/svg">
      <path
        className="wu-card-progressbg"
        d="M 6 2 H 194 Q 198 2,198 6 V 314 Q 198 318,194 318 H 6 Q 2 318,2 314 V 6 Q 2 2,6 2 z" />
      <path
        className="wu-card-border"
        d="M 6 2 H 194 Q 198 2,198 6 V 314 Q 198 318,194 318 H 6 Q 2 318,2 314 V 6 Q 2 2,6 2 z" />
    </svg>
    <div
      className="wu-card-content" />
  </div>
  <div
    className="wu-card-box ">
    <svg
      className="wu-card-bg"
      height={320}
      width={200}
      xmlns="http://www.w3.org/2000/svg">
      <path
        className="wu-card-progressbg"
        d="M 6 2 H 194 Q 198 2,198 6 V 314 Q 198 318,194 318 H 6 Q 2 318,2 314 V 6 Q 2 2,6 2 z" />
      <path
        className="wu-card-border"
        d="M 6 2 H 194 Q 198 2,198 6 V 314 Q 198 318,194 318 H 6 Q 2 318,2 314 V 6 Q 2 2,6 2 z" />
    </svg>
    <div
      className="wu-card-content">
      <span
        onClick={[Function onClick]}>
        Add new
      </span>
    </div>
  </div>
</div>
`;
